feat(autoware_pointcloud_preprocessor): update the readme file and schema file

vish0012 opened the pull request:
## Description
Implement the ROS Node configuration layout described in https://github.com/orgs/autowarefoundation/discussions/3371 for the autoware_pointcloud_preprocessor package.

- update json schema
- update readme file

## Related links

**Parent Issue:**

- Link

<!-- ⬇️🟢
**Private Links:**

- [CompanyName internal link]()
⬆️🟢 -->

## How was this PR tested?

- colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release --packages-up-to autoware_pointcloud_preprocessor
- ros2 launch launch/autoware_pointcloud_preprocessor.launch.xml

## Notes for reviewers

None.

## Interface changes

None.

<!-- ⬇️🔴

### Topic changes

#### Additions and removals

| Change type   | Topic Type      | Topic Name    | Message Type        | Description       |
|:--------------|:----------------|:--------------|:--------------------|:------------------|
| Added/Removed | Pub/Sub/Srv/Cli | `/topic_name` | `std_msgs/String`   | Topic description |

#### Modifications

| Version | Topic Type      | Topic Name        | Message Type        | Description       |
|:--------|:----------------|:------------------|:--------------------|:------------------|
| Old     | Pub/Sub/Srv/Cli | `/old_topic_name` | `sensor_msgs/Image` | Topic description |
| New     | Pub/Sub/Srv/Cli | `/new_topic_name` | `sensor_msgs/Image` | Topic description |

### ROS Parameter Changes

#### Additions and removals

| Change type   | Parameter Name | Type     | Default Value | Description       |
|:--------------|:---------------|:---------|:--------------|:------------------|
| Added/Removed | `param_name`   | `double` | `1.0`         | Param description |

#### Modifications

| Version | Parameter Name   | Type     | Default Value | Description       |
|:--------|:-----------------|:---------|:--------------|:------------------|
| Old     | `old_param_name` | `double` | `1.0`         | Param description |
| New     | `new_param_name` | `double` | `1.0`         | Param description |

🔴⬆️ -->

## Effects on system behavior

None.

ambroise-arm commented:
https://autowarefoundation.github.io/autoware.universe/pr-10193/sensing/autoware_pointcloud_preprocessor/ shows "Macro Rendering Error", is that expected?

mojomex commented on sensing/autoware_pointcloud_preprocessor/schema/pickup_based_voxel_grid_downsample_filter_node.schema.json:
```diff
@@ -1,46 +1,39 @@
-      "required": ["ros__parameters"],
-      "additionalProperties": false
+    "ros__parameters": {
+      "$ref": "#/definitions/pickup_based_voxel_grid_downsample_filter_node"
     }
   },
-  "required": ["/**"],
+  "required": ["ros__parameters"],
```
This should stay the same as before, otherwise it will not be read correctly by ROS or the markdown generator.

vish0012 commented on sensing/autoware_pointcloud_preprocessor/schema/passthrough_filter_uint16_node.schema.json:
```diff
@@ -45,17 +45,10 @@
   },
-  "required": ["/**"],
+  "required": ["ros__parameters"],
   "additionalProperties": false
```
done thank you

vish0012 commented on sensing/autoware_pointcloud_preprocessor/schema/pickup_based_voxel_grid_downsample_filter_node.schema.json:
```diff
@@ -1,46 +1,39 @@
-          "exclusiveMinimum": 0.0
+          "description": "Voxel size along the x-axis [m].",
+          "default": 1.0,
+          "minimum": 0.0
```
done thank you

vish0012 commented on sensing/autoware_pointcloud_preprocessor/schema/blockage_diag_node.schema.json:
```diff
@@ -1,93 +1,93 @@
-          "description": "enable dust diagnostic",
-          "default": "false"
+          "description": "Enable dust diagnostic",
+          "default": false
```
done thank you

mojomex commented on sensing/autoware_pointcloud_preprocessor/README.md:
```diff
@@ -43,16 +43,23 @@ Detail description of each filter's algorithm is in the following links.
-| `use_indices`      | bool   | false         | flag to use pointcloud indices        |
-| `latched_indices`  | bool   | false         | flag to latch pointcloud indices      |
-| `approximate_sync` | bool   | false         | flag to use approximate sync option   |
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/approximate_downsample_filter_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/blockage_diag_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/concatenate_pointclouds.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/crop_box_filter_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/distortion_corrector_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/dual_return_outlier_filter_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/lanelet2_map_Filter_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/passthrough_filter_uint16_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/pickup_based_voxel_grid_downsample_filter_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/pointcloud_accumulator_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/radius_search_2d_outlier_filter_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/random_downsample_filter_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/ring_outlier_filter_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/time_synchronizer_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/vector_map_inside_area_filter_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/voxel_grid_downsample_filter_node.schema.json") }}
+{{ json_to_markdown("sensing/autoware_pointcloud_preprocessor/schema/voxel_grid_outlier_filter_node.schema.json") }}
```
Thank you for addressing the other changes. This conversation here has been closed without being addressed, so I reopened it. All those JSON tables are already found in the pages linked in `README.md`, so there is no particular value in adding them here as well. Also, there's still rendering errors: https://autowarefoundation.github.io/autoware_universe/pr-10193/sensing/autoware_pointcloud_preprocessor/

Please address those first before re-requesting a review, thank you :pray:

technolojin commented:
![Screenshot from 2025-06-18 15-04-00](https://github.com/user-attachments/assets/adbd6f94-61c5-4e6c-adf4-032ca4d67ed8)

the documentation is broken. it need to be fixed.

mojomex commented:
@vish0012 Hi, I'd like to proceed with reviewing/merging this PR, but there is still some unaddressed feedback. Could you have a look at the comments when you find the time? :bow:

vish0012 commented:
> @vish0012 Hi, I'd like to proceed with reviewing/merging this PR, but there is still some unaddressed feedback. Could you have a look at the comments when you find the time? 🙇

>@mojomex "Thank you for your comments. I’ll review feedback again and address anything I may have missed."

vish0012 commented:
> > @vish0012 Hi, I'd like to proceed with reviewing/merging this PR, but there is still some unaddressed feedback. Could you have a look at the comments when you find the time? 🙇
> 
> > @mojomex "Thank you for your comments. I’ll review feedback again and address anything I may have missed."

> > @mojomex "Thank you, i have solved the conflict files issues , could you let me know if something left.
